src/test/scala/io/iohk/ethereum/vm/CallOpFixture.scala

```diff
@@ -0,0 +1,157 @@
+    PUSH1, 2,
+    CALLDATASIZE,
+    DIV,
+    PUSH1, 0,
+    DUP2,
+    DUP2,
+    DUP1,
+    CALLDATACOPY,
+    RETURN
+  )
+
+  val selfDestructCode = Assembly(
+    PUSH20, callerAddr.bytes,
+    SELFDESTRUCT
+  )
+
+  val selfDestructTransferringToSelfCode = Assembly(
+    PUSH20, extAddr.bytes,
+    SELFDESTRUCT
+  )
+
+  val sstoreWithClearCode = Assembly(
+    //Save a value to the storage
+    PUSH1, 10,
+    PUSH1, 0,
+    SSTORE,
+
+    //Clear the store
+    PUSH1, 0,
+    PUSH1, 0,
+    SSTORE
+  )
+
+  val valueToReturn = 23
+  val returnSingleByteProgram = Assembly(
+    PUSH1, valueToReturn,
+    PUSH1, 0,
+    MSTORE,
+    PUSH1, 1,
+    PUSH1, 31,
+    RETURN
+  )
+
+  val inputData = Generators.getUInt256Gen().sample.get.bytes
+  val expectedMemCost = config.calcMemCost(inputData.size, inputData.size, inputData.size / 2)
+
+  val initialBalance = UInt256(1000)
+
+  val requiredGas = {
+    val storageCost = 3 * G_sset
+    val memCost = config.calcMemCost(0, 0, 32)
+    val copyCost = G_copy * wordsForBytes(32)
+
+    extCode.linearConstGas(config) + storageCost + memCost + copyCost
+  }
+
+  val gasMargin = 13
+
+  val initialOwnerAccount = Account(balance = initialBalance)
+
+  val extProgram = extCode.program
+  val invalidProgram = Program(extProgram.code.init :+ INVALID.code)
+  val selfDestructProgram = selfDestructCode.program
+  val sstoreWithClearProgram = sstoreWithClearCode.program
+  val accountWithCode: ByteString => Account = code => Account.empty().withCode(kec256(code))
+
+  val worldWithoutExtAccount = startState.saveAccount(ownerAddr, initialOwnerAccount)
+
+  val worldWithExtAccount = worldWithoutExtAccount.saveAccount(extAddr, accountWithCode(extProgram.code))
+    .saveCode(extAddr, extProgram.code)
+
+  val worldWithExtEmptyAccount = worldWithoutExtAccount.saveAccount(extAddr, Account.empty())
+
+  val worldWithInvalidProgram = worldWithoutExtAccount.saveAccount(extAddr, accountWithCode(invalidProgram.code))
+    .saveCode(extAddr, invalidProgram.code)
+
+  val worldWithSelfDestructProgram = worldWithoutExtAccount.saveAccount(extAddr, accountWithCode(selfDestructProgram.code))
+    .saveCode(extAddr, selfDestructCode.code)
+
+  val worldWithSelfDestructSelfProgram = worldWithoutExtAccount.saveAccount(extAddr, Account.empty())
+    .saveCode(extAddr, selfDestructTransferringToSelfCode.code)
+
+  val worldWithSstoreWithClearProgram = worldWithoutExtAccount.saveAccount(extAddr, accountWithCode(sstoreWithClearProgram.code))
+    .saveCode(extAddr, sstoreWithClearCode.code)
+
+  val worldWithReturnSingleByteCode = worldWithoutExtAccount.saveAccount(extAddr, accountWithCode(returnSingleByteProgram.code))
+    .saveCode(extAddr, returnSingleByteProgram.code)
+
+  val env = ExecEnv(ownerAddr, callerAddr, callerAddr, 1, ByteString.empty, 123, Program(ByteString.empty), null, 0)
+  val context: PC = ProgramContext(env, ownerAddr, 2 * requiredGas, worldWithExtAccount, config)
+
+  case class CallResult(
```

Comment:
I like this separation. One thing though: does it need to be a path-dependent type? If not, it would better to move it to `CallOpFixture` companion and import it from there.

Reply:
with the current design, I am afraid it needs to be as all default parameters depend on values in  `CallOpFixture`. Maybe we would refactor this in a separate task as I don't like the idea of messing with those tests fixture more than necessary or duplicating `CallResult` in two test classes .

Reply:
>  default parameters depend on values in CallOpFixture

Of course.. That's ok then, nevermind